### PR TITLE
NUT-06: add urls field

### PR DESCRIPTION
**Pull request by prusnak**
fixes https://github.com/cashubtc/nuts/issues/62

- [x] CDK - https://github.com/cashubtc/cdk/pull/398
- [x] Nutshell - https://github.com/cashubtc/nutshell/pull/638

### CODE REVIEW

File: 06.md [hunk 1]
```diff
@@ -39,6 +39,10 @@ With the mint's response being of the form `GetInfoResponse`:
   ],
   "motd": "Message to display to users.",
   "icon_url": "https://mint.host/icon.jpg",
+  "urls": [
```
Comment (prusnak) [inline 06.md:42]: should we call the field `urls` or `endpoints` or something else?

**Comment by thesimplekid** [comment]
I think we should not just list urls but follow a struct more similar to the contact feild for examples:

```json 
{
  "endpoints": [
    {
      "method": "http",
      "endpoint": "https://fake.mint.dev"
    },
    {
      "method": "nostr",
      "endpoint": "nprofile1qs..."
    },
    {
      "method": "tor",
      "endpoint": "abcdefghijklmn.onion"
    }
  ]
}

```

**Comment by prusnak** [comment]
> I think we should not just list urls but follow a struct more similar to the contact feild for examples:

🤷 URI schemes were invented exactly for this reason.I am just not a big fan of using dicts and I think it would have been much better if contact field also used URI schemes instead of the JSON dict abomination we have now. :-) That said I don't mind either way and I prefer consistency over correctness. If there is consensus to go with the JSON dict instead of the URI scheme, I will update the PR.

**Comment by ok300** [comment]
`TokenV4` has a field `m` for the mint URL: https://cashubtc.github.io/nuts/00/#token-format_1

Consider the following scenario:

Let's say mint Bob in accessible on 2 URLs. If Alice has a token for Bob's mint (with "Bob URL 1" embedded) and sends this token to Carol, then Carol will only have "Bob URL 1" to get to the mint. If the mint becomes unreachable on URL 1 but remains reachable on it's URL 2, Carol won't know the 2nd URL and won't be able to redeem it.

Is that true, or am I missing something?

The fix would be for `TokenV4` to include all of Bob's known URLs, but that would mean making the token bigger (esp. with tor URLs).

**Comment by prusnak** [comment]
> Consider the following scenario:

That is exactly what I described here:

https://github.com/cashubtc/nuts/issues/62#issuecomment-2404500881

**Comment by thesimplekid** [comment]
Merges in cdk https://github.com/cashubtc/cdk/pull/398

**Comment by prusnak** [comment]
Merged in nutshell https://github.com/cashubtc/nutshell/pull/638

**Review by thesimplekid — APPROVED** [review]

**Pull request merged**